src/slotshandler.cpp

```diff
@@ -103,6 +103,12 @@ int file_size (int file_num)
    }
 }
 
+#define FILL_SLOT_INFO(filevar, processedvar, file) \
```

Comment:
You can do the same (but with the benefit of a function and without the issues of macros) with:
void fill_slot_info(std::string &filevar, bool &processedvar, const string& fullpath)
{
  filevar = fullpath;
  processedvar = true;
}

You could also add the extension and factorize the if & logging part.